central/debug/service/service.go

```diff
@@ -343,6 +373,95 @@ func (s *serviceImpl) getLogImbue(zipWriter *zip.Writer) error {
 	return err
 }
 
+func (s *serviceImpl) getAuthProviders(_ context.Context) (interface{}, error) {
+	authProviders := s.authProviderRegistry.GetProviders(nil, nil)
+
+	var storageAuthProviders []*storage.AuthProvider
+	for _, authProvider := range authProviders {
+		storageAuthProviders = append(storageAuthProviders, authProvider.StorageView())
+	}
+
+	return storageAuthProviders, nil
+}
+
+func (s *serviceImpl) getGroups(ctx context.Context) (interface{}, error) {
+	// We will elevate the user rights in order to get all groups
+	accessGroupsCtx := sac.WithGlobalAccessScopeChecker(ctx,
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
+			sac.ResourceScopeKeys(resources.Group)))
+
+	return s.groupDataStore.GetAll(accessGroupsCtx)
+}
+
+type diagResolvedRole struct {
+	Role          *storage.Role              `json:"role,omitempty"`
+	PermissionSet map[string]string          `json:"permission_set,omitempty"`
+	AccessScope   *storage.SimpleAccessScope `json:"access_scope,omitempty"`
+}
+
+func (s *serviceImpl) getRoles(ctx context.Context) (interface{}, error) {
+	// We will elevate the user rights in order to get all roles
+	accessRolesCtx := sac.WithGlobalAccessScopeChecker(ctx,
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
+			sac.ResourceScopeKeys(resources.Role)))
+
+	roles, errGetRoles := s.roleDataStore.GetAllRoles(accessRolesCtx)
+	if errGetRoles != nil {
+		return nil, errGetRoles
+	}
+
+	var resolvedRoles []*diagResolvedRole
+	for _, role := range roles {
+		diagRole := diagResolvedRole{
+			Role: role,
+		}
+
+		resolvedRole, errGetResolved := s.roleDataStore.GetAndResolveRole(accessRolesCtx, role.Name)
+		if errGetResolved == nil {
+			// Get better formatting of permission sets
+			diagRole.PermissionSet = map[string]string{}
+			for permName, accessRight := range resolvedRole.GetPermissions() {
+				diagRole.PermissionSet[permName] = accessRight.String()
+			}
+			diagRole.AccessScope = resolvedRole.GetAccessScope()
+		}
+
+		resolvedRoles = append(resolvedRoles, &diagRole)
+	}
+
+	return resolvedRoles, nil
+}
+
+func (s *serviceImpl) getNotifiers(ctx context.Context) (interface{}, error) {
+	// We will elevate the user rights in order to get notifiers
+	accessNotifierCtx := sac.WithGlobalAccessScopeChecker(ctx,
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
+			sac.ResourceScopeKeys(resources.Notifier)))
+
+	notifiers, err := s.notifierDataStore.GetNotifiers(accessNotifierCtx)
+	if err != nil {
+		return nil, err
+	}
+	for _, n := range notifiers {
+		secrets.ScrubSecretsFromStructWithReplacement(n, secrets.ScrubReplacementStr)
+	}
+
+	return notifiers, nil
+}
+
+func (s *serviceImpl) getConfig(ctx context.Context) (interface{}, error) {
+	// We will elevate the user rights in order to get config
+	accessConfigCtx := sac.WithGlobalAccessScopeChecker(ctx,
```

Comment:
To be pedantic, `context.Background()` is probably cleaner than `ctx`. The latter can have all sorts of keys and theoretically lead to different outcomes for different requests. Here and above.

If you follow this suggestion, I'd recommend to keep context in the signature and explicitly ignore it, `getConfig(_ context.Context)`. I believe this way we're communicating to the callers that while the fetched data are access controlled, we override the access checks here.